packages/bpk-component-chip-group/src/BpkChipGroup.tsx

```diff
@@ -0,0 +1,213 @@
+export type ChipStyleType = (typeof CHIP_TYPES)[keyof typeof CHIP_TYPES];
+export type ChipComponentType = (typeof CHIP_COMPONENT)[keyof typeof CHIP_COMPONENT];
+
+export type SingleSelectChipItem = {
+  text: string;
+  accessibilityLabel?: string;
+  leadingAccessoryView?: ReactNode;
+  [rest: string]: any; // Inexact rest. See decisions/inexact-rest.md
+};
+
+export type ChipItem = {
+  component?: ChipComponentType;
+  onClick?: (selected: boolean, index: number) => void;
+  selected?: boolean;
+  hidden?: boolean;
+} & SingleSelectChipItem;
+
+export type CommonProps = {
+  ariaLabel?: string;
+  type?: ChipGroupType;
+  chipStyle?: ChipStyleType;
+  // only required when type = rail
+  leadingNudgerLabel?: string;
+  trailingNudgerLabel?: string;
+};
+
+export type ChipGroupProps = {
+  chips: ChipItem[];
+  stickyChip?: ChipItem;
+  ariaMultiselectable?: boolean;
+} & CommonProps;
+
+
+const BpkChipGroup = ({
+  ariaLabel,
+  ariaMultiselectable = true,
+  chipStyle = CHIP_TYPES.default,
+  chips,
+  leadingNudgerLabel = '',
+  stickyChip,
+  trailingNudgerLabel = '',
+  type = CHIP_GROUP_TYPES.rail,
+}: ChipGroupProps) => {
+  const scrollContainerRef = useRef<HTMLElement | null>(null);
+
+  const containerClassnames = getClassName('bpk-chip-group-container')
+
+  const chipGroupClassNames = getClassName(
+    'bpk-chip-group',
+    `bpk-chip-group--${type}`,
+  );
+
+  const stickyChipContainerClassnames = getClassName(
+    'bpk-sticky-chip-container',
+    `bpk-sticky-chip-container--${chipStyle}`,
+  );
+
+  const scrollContainerIndicator = getClassName('bpk-chip-group-scroller-indicator');
+
+  const renderChipItem = ({
+    accessibilityLabel,
+    component = CHIP_COMPONENT.selectable,
+    hidden = false,
+    leadingAccessoryView = null,
+    onClick,
+    selected,
+    text,
+    ...rest
+  }: ChipItem, index: number) => {
+    const Component = CHIP_COMPONENT_MAP[component];
+    return !hidden && (
+      <Component
+        key={text}
+        selected={selected ?? false}
+        type={chipStyle}
+        accessibilityLabel={accessibilityLabel || text}
+        onClick={() => {
+          if (onClick) {
+            onClick(!selected, index);
+          }
+        }}
+        role={ariaMultiselectable ? 'checkbox' : 'radio'}
+        leadingAccessoryView={leadingAccessoryView}
+        {...rest}
+      >
+        {text}
+      </Component>
+    );
+  }
+
+  const wrapRailInScroll = (children: ReactElement) =>
+    type === CHIP_GROUP_TYPES.rail ? (
+      <BpkMobileScrollContainer
+        scrollerRef={(el: HTMLElement) => {scrollContainerRef.current = el}}
+        leadingIndicatorClassName={scrollContainerIndicator}
+        trailingIndicatorClassName={scrollContainerIndicator}
+      >
+        {children}
+      </BpkMobileScrollContainer>
+    ) : children;
+
+  return (
+    <div className={containerClassnames}>
+      {type === CHIP_GROUP_TYPES.rail && (
```

Comment:
`type === CHIP_GROUP_TYPES.rail` there are a lot of these statements. Have you considered if the different types of `BpkChipGroup` are more different to one another than they are similar? 

Basically, did you look at having:

```tsx
const BpkChipGroup = ({type, ...rest}) => {
    if(type === "rail"){
        return <RailChipGroup {...rest}/>
   }

    return <DefaultChipGroup {...rest} />
}

```